tests/queryset_pickle/tests.py

```diff
@@ -115,6 +115,38 @@ def test_pickle_prefetch_related_idempotence(self):
         groups = pickle.loads(pickle.dumps(groups))
         self.assertSequenceEqual(groups, [g])
 
+    def test_pickle_prefetch_queryset_not_evaluated(self):
```

Comment:
I think separate commits makes sense though these tests check for regressions in the initial implementation on this PR so I'd say they count as tests for the current change